tests/Templating/TemplateRegistryTest.php

```diff
@@ -25,63 +25,35 @@ class TemplateRegistryTest extends TestCase
 
     protected function setUp(): void
     {
-        $this->templateRegistry = new TemplateRegistry();
+        $this->templateRegistry = new TemplateRegistry([
+            'list' => '@FooAdmin/CRUD/list.html.twig',
+            'show' => '@FooAdmin/CRUD/show.html.twig',
+            'edit' => '@FooAdmin/CRUD/edit.html.twig',
+        ]);
     }
 
     public function testGetTemplates(): void
     {
-        $this->assertSame([], $this->templateRegistry->getTemplates());
-
         $templates = [
             'list' => '@FooAdmin/CRUD/list.html.twig',
             'show' => '@FooAdmin/CRUD/show.html.twig',
             'edit' => '@FooAdmin/CRUD/edit.html.twig',
         ];
 
-        $this->templateRegistry->setTemplates($templates);
         $this->assertSame($templates, $this->templateRegistry->getTemplates());
     }
 
     /**
      * @group legacy
      *
-     * @expectedDeprecation Passing a nonexistent template name as argument 1 to Sonata\AdminBundle\Templating\TemplateRegistry::getTemplate() is deprecated since sonata-project/admin-bundle 3.52 and will throw an exception in 4.0.
+     * @expectedDeprecation Passing a nonexistent template name as argument 1 to Sonata\AdminBundle\Templating\AbstractTemplateRegistry::getTemplate() is deprecated since sonata-project/admin-bundle 3.52 and will throw an exception in 4.0.
```

Comment:
Please, use `ExpectDeprecationTrait::expectDeprecation()` instead and update the `NEXT_MAJOR` comment inside the method.

Reply:
I can't :cry: :
`Failed asserting that exception of type "PHPUnit\Framework\Error\Deprecated" is thrown.`

Reply:
Did you add `use ExpectDeprecationTrait;` ?

Reply:
I used `PHPUnit\Framework\TestCase::expectDeprecation`